### PR TITLE
[SPARK-32313][PYSPARK] Remove Python 2 artifacts

### PR DESCRIPTION
**Pull request by Fokko**
@HyukjinKwon Some lingering Python 2 artifacts that can be removed.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
ok to test

File: python/pyspark/cloudpickle.py [hunk 1]
```diff
@@ -1010,11 +1006,7 @@ def save_ellipsis(self, obj):
     def save_not_implemented(self, obj):
         self.save_reduce(_gen_not_implemented, ())
 
-    try:               # Python 2
-        dispatch[file] = save_file
-    except NameError:  # Python 3  # pragma: no branch
-        dispatch[io.TextIOWrapper] = save_file
-
```
Comment (HyukjinKwon) [inline python/pyspark/cloudpickle.py:1017]: Ah, let's match cloudpickle with the released version. That's why I didnt touch this code.

File: python/pyspark/_globals.py [hunk 2]
```diff
@@ -59,10 +59,6 @@ def __new__(cls):
             cls.__instance = super(_NoValueType, cls).__new__(cls)
         return cls.__instance
 
-    # needed for python 2 to preserve identity through a pickle
-    def __reduce__(self):
-        return (self.__class__, ())
-
```
Comment (HyukjinKwon) [inline python/pyspark/_globals.py:65]: This one I feel less strongly but I didnt touch because this class was directly ported from NumPy.

Comment (Fokko) [reply python/pyspark/_globals.py:65]: Thanks for clearing up, lets keep it for now.

File: python/pyspark/cloudpickle.py (hunk 1 above)
Comment (Fokko) [reply python/pyspark/cloudpickle.py:1017]: Good point.

**Pull request closed without merging**